modules/stanford_earth_capx/src/EarthCapxInfo.php

```diff
@@ -0,0 +1,249 @@
+
+  const EARTH_CAPX_INFO_TABLE = 'migrate_info_earth_capx_importer';
+
+  private $sunetid;
+  private $etag;
+  private $profilePhotoTimestamp;
+  private $entity_id;
+  private $profilePhotoFid;
+  private $status;
+
+  /**
+   * construct object with sunetid parameter;
+   * if already in table, populate other properties.
+   *
+   * @param $su_id
+   */
+  public function __construct($su_id = "") {
+    $su_id = (string) $su_id;
+    $this->sunetid = "";
+    $this->status = self::EARTH_CAPX_INFO_INVALID;
+    $this->etag = "";
+    $this->profilePhotoTimestamp = "";
+    $this->entity_id = 0;
+    $this->profilePhotoFid = 0;
+    if (!empty($su_id)) {
+      $this->status = self::EARTH_CAPX_INFO_NEW;
+      $this->sunetid = $su_id;
+      $db = \Drupal::database();
+      $result = $db->query("SELECT * FROM {" . self::EARTH_CAPX_INFO_TABLE .
+        "} WHERE sunetid = :sunetid", [ ':sunetid' => $this->sunetid]);
+      foreach ($result as $record) {
+        $this->status = self::EARTH_CAPX_INFO_FOUND;
+        if (!empty($record->etag)) {
+          $this->etag = $record->etag;
+        }
+        if (!empty($record->photo_timestamp)) {
+          $this->profilePhotoTimestamp = $record->photo_timestamp;
+        }
+        if (!empty($record->entity_id)) {
+          $this->entity_id = intval($record->entity_id);
+        }
+        if (!empty($record->profile_photo_id)) {
+          $this->profilePhotoFid = intval($record->profile_photo_id);
+        }
+      }
+    }
+  }
+
+  /**
+   * get the photo timestamp from the cap url
+   */
+  private function getPhotoTimestamp($photoUrl = "") {
+    $photo_ts = '';
+    if (!empty($photoUrl) && is_string($photoUrl)) {
+      $ts1 = strpos($photoUrl,"ts=");
+      if ($ts1 !== false) {
+        $ts2 = substr($photoUrl,$ts1);
+        if (strpos($ts2,"&") !== false)  {
+          $photo_ts = substr($ts2,3,strpos($ts2,"&")-3);
+        } else {
+          $photo_ts = substr($ts2,3);
+        }
+      }
+    }
+    return $photo_ts;
+  }
+  
+  /**
+   * see if we have an existing and current profile photo_id
+   * if we do, return its fid. if we don't, return false
+   * @param array $source
+   */
+  public function currentProfilePhotoId($source = []) {
+    if (empty($source['profile_photo']) || empty($this->profilePhotoFid)
+      || empty($this->profilePhotoTimestamp)) {
+      return false;
+    }
+    $photo_ts = $this->getPhotoTimestamp($source['profile_photo']);
+    if ($photo_ts == $this->profilePhotoTimestamp) {
+      return $this->profilePhotoFid;
+    } else {
+      return false;
+    }
+  }
+  
+  /**
+   * return true if profile should be updated because it is new
+   * or because the etag has changed.
+   * @param array $source
+   */
+  public function getOkayToUpdateProfile($source = [], $photoId = 0) {
+    $oktoupdate = false;
+    $msg = new MigrateMessage();
+    if (empty($source['sunetid']) ||
+      $this->status == self::EARTH_CAPX_INFO_INVALID ||
```

Comment:
It would be good to document here where and how $this->status gets set.

Reply:
Done.